databuilder/extractor/athena_metadata_extractor.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from collections import namedtuple
+
+from pyhocon import ConfigFactory, ConfigTree  # noqa: F401
+from typing import Iterator, Union, Dict, Any  # noqa: F401
+
+from databuilder import Scoped
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.sql_alchemy_extractor import SQLAlchemyExtractor
+from databuilder.models.table_metadata import TableMetadata, ColumnMetadata
+from itertools import groupby
+
+TableKey = namedtuple('TableKey', ['schema_name', 'table_name'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AthenaMetadataExtractor(Extractor):
+    """
+    Extracts Postgres table and column metadata from underlying meta store database using SQLAlchemyExtractor
+    """
+
+    SQL_STATEMENT = """
+    SELECT
```

Comment:
the query seems to be very similar to the postgres extractor(https://github.com/lyft/amundsendatabuilder/blob/master/databuilder/extractor/postgres_metadata_extractor.py#L25) . Is Athena the AWS version of postgres?

cc @jinhyukchang @p5k6

Reply:
No - it's actually presto based, using an aws-provided hive metastore that is compatible with glue, athena and EMR (hive, spark, presto, etc on EMR). The tricky part is - I'm not sure if they have any "aws special sauce" in their metastore.

Regarding the `information_schema` piece though - that's an ANSI standard for dbs. Apparently it's supported by presto (my experience with presto is pretty limited, but [github seems to have confirmed this](https://github.com/nileema/presto/commit/0c9ba26e66961382832a4f806247031271473b4c)). Queries using `information_schema` pretty easily port from one db to another.

Though we should probably change the comment about "extracts postgres table and column metadata" to Athena on line 20 😄

Reply:
@feng-tao  not really, as @p5k6 wrote it's basically presto.
Using the `information_schema` is the easiest way to get the needed data of the tables. 
One important thing is the partition column which is the column you must use when querying with athena unless you want to pay a lot for each query $$$. Athena saves it under `extra_info` column of `information_schema.columns`

It would be super interesting of what kind of data we can pull from Glue - usage maybe?

Reply:
Yeah, getting usage data somewhere(e.g from Glue) will be very useful as the search uses usage data to do the ranking.